src/Experiment/tests/ReflectionTesting.cs

```diff
@@ -0,0 +1,465 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.CodeDom.Compiler;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit.Experimental;
+using System.Reflection.Emit.Experimental.Tests;
+using Xunit;
+
+namespace Experiment.Tests
+{
+    //interfaces for testing
+    public interface INoMethod
+    {
+
+    }
+
+    public interface INoMethod2
+    {
+
+    }
+
+    public interface IMultipleMethod
+    {
+        string Func(int a, string b);
+        bool MoreFunc(int a, string b, bool c);
+        bool DoIExist();
+        void BuildAPerpetualMotionMachine();
+    }
+
+    internal interface IAccess
+    {
+        public void BuildAI();
+        public int DisableRogueAI();
+    }
+
+    public interface IOneMethod
+    {
+        string Func(int a, string b);
+    }
+
+    public class ReflectionTesting
+    {
+        const bool _keepFiles = true; // keep files after testing for inspection
+        TempFileCollection _tfc;
+
+        [Fact]
```

Comment:
I mentioned this previously. This is not the correct way to setup a test class asset. See https://xunit.net/docs/shared-context